lighthouse/tests/validator_client.rs

```diff
@@ -0,0 +1,337 @@
+        self
+    }
+
+    fn run(&mut self) -> CompletedTest {
+        // Setup temp directories.
+        let tmp_dir = TempDir::new().expect("Unable to create temporary directory");
+        let tmp_path: PathBuf = tmp_dir.path().join(CONFIG_NAME);
+
+        // Add --datadir <temp_dir> --dump-config <temp_path> to cmd.
+        self.cmd
+            .arg("--datadir")
+            .arg(tmp_dir.path().as_os_str())
+            .arg(format!("--{}", DUMP_CONFIG_CMD))
+            .arg(tmp_path.as_os_str());
+
+        // Run the command.
+        let _output = output_result(&mut self.cmd).expect("Unable to run command");
+
+        // Grab the config.
+        let config: Config =
+            from_reader(File::open(tmp_path).expect("Unable to open dumped config"))
+                .expect("Unable to deserialize to ClientConfig");
+        CompletedTest {
+            config,
+            dir: tmp_dir,
+        }
+    }
+
+    // In order to test custom validator and secrets directory flags,
+    // datadir cannot be defined.
+    fn run_with_no_datadir(&mut self) -> CompletedTest {
+        // Setup temp directories
+        let tmp_dir = TempDir::new().expect("Unable to create temporary directory");
+        let tmp_path: PathBuf = tmp_dir.path().join(CONFIG_NAME);
+
+        // Add --dump-config <temp_path> to cmd.
+        self.cmd
+            .arg(format!("--{}", DUMP_CONFIG_CMD))
+            .arg(tmp_path.as_os_str());
+
+        // Run the command.
+        let _output = output_result(&mut self.cmd).expect("Unable to run command");
+
+        // Grab the config.
+        let config: Config =
+            from_reader(File::open(tmp_path).expect("Unable to open dumped config"))
+                .expect("Unable to deserialize to ClientConfig");
+        CompletedTest {
+            config,
+            dir: tmp_dir,
+        }
+    }
+}
+struct CompletedTest {
+    config: Config,
+    dir: TempDir,
+}
+impl CompletedTest {
+    fn with_config<F: Fn(&Config)>(self, func: F) {
+        func(&self.config);
+    }
+    fn with_config_and_dir<F: Fn(&Config, &TempDir)>(self, func: F) {
+        func(&self.config, &self.dir);
+    }
+}
+
+#[test]
+fn datadir_flag() {
+    CommandLineTest::new()
+        .run()
+        .with_config_and_dir(|config, dir| {
+            assert_eq!(config.validator_dir, dir.path().join("validators"));
+            assert_eq!(config.secrets_dir, dir.path().join("secrets"));
+        });
+}
+
+#[test]
+fn validators_and_secrets_dir_flags() {
+    let dir = TempDir::new().expect("Unable to create temporary directory");
+    CommandLineTest::new()
+        .flag("validators-dir", dir.path().join("validators").to_str())
+        .flag("secrets-dir", dir.path().join("secrets").to_str())
+        .run_with_no_datadir()
+        .with_config(|config| {
+            assert_eq!(config.validator_dir, dir.path().join("validators"));
+            assert_eq!(config.secrets_dir, dir.path().join("secrets"));
+        });
+}
+
```

Comment:
Would you mind also adding a test that checks the default `boot-nodes` value? I know we don't usually check the default values, but this one is important and I'd love to get an extra warning if it changes :)